incubator/hnc/internal/validators/object_test.go

```diff
@@ -308,6 +308,79 @@ func TestUserChanges(t *testing.T) {
 				},
 			},
 		},
+	}, {
+		name: "Deny creation of object with invalid treeSelect annotation",
+		fail: true,
+		inst: &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "v1",
+				"kind":       "Pod",
+				"metadata": map[string]interface{}{
+					"annotations": map[string]interface{}{
+						api.AnnotationTreeSelector: "foo, $bar",
+					},
+				},
+			},
+		},
+	}, {
+		name: "Allow creation of object with valid treeSelect annotation",
+		fail: false,
+		inst: &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "v1",
+				"kind":       "Pod",
+				"metadata": map[string]interface{}{
+					"annotations": map[string]interface{}{
+						api.AnnotationTreeSelector: "foo, !bar",
+					},
+				},
+			},
+		},
+	}, {
+		name: "Deny creation of object with invalid selector and valid treeSelect annotation",
+		fail: true,
+		inst: &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "v1",
+				"kind":       "Pod",
+				"metadata": map[string]interface{}{
+					"annotations": map[string]interface{}{
+						api.AnnotationSelector:     "$foo",
+						api.AnnotationTreeSelector: "!bar",
+					},
+				},
+			},
+		},
+	}, {
+		name: "Deny creation of object with valid selector and invalid treeSelect annotation",
+		fail: true,
+		inst: &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "v1",
+				"kind":       "Pod",
+				"metadata": map[string]interface{}{
+					"annotations": map[string]interface{}{
+						api.AnnotationSelector:     "foo",
+						api.AnnotationTreeSelector: "$bar",
+					},
+				},
+			},
+		},
+	}, {
+		name: "Allow creation of object with valid selector and treeSelect annotation",
```

Comment:
I thought we were going to disallow this? But if we haven't implemented it yet, that's fine for this PR

Reply:
Ah right. When it's implemented I'll update this test

Reply:
Just confirming that we decided to only allow one type of selector at a time and cannot mix any two, right?

Reply:
Yup!